File: CCC-school-work/comporg/pnc2/ahought2/lua/primeregbs.lua
#!/usr/bin/lua5.3

-- comments like this

--[[multi line
comments like this]]

number = 1 
count  = 0 
factor = 2 
check  = 1

--[[if arg[1] then
    print("Error: invalid arguments\n")
else]]
qty = tonumber(arg[1])
--end

local time = os.clock()
while (count < qty) do
    number = number + 1
    num = math.sqrt(number) --find sqrt / can be decimal
    sqrt = math.floor(num)  --find the largest int smaller than or equal to (drops decimal)
    check = 1
    factor = 2
    while (factor <= sqrt) do
        mod = math.fmod(number,factor)
        if mod == 0 then
            check = 0
            break
        end
        factor = factor + 1
    end
    if check == 1 then
        io.stderr:write(number, " " )
        count = count + 1
    end
end
print()
print(string.format("%8.4f", os.clock() - time)) 

File: CCC-school-work/comporg/pnc2/ahought2/ruby/primeregbs.rb
#!/usr/bin/ruby

# comments are using the pound key (hashtag for the young kids)
if ARGV.empty?
    puts "Error: No quantity given!\n"
    exit
end

number = 1
factor = 2
count  = 0
check  = 1
qty    = ARGV[0].to_i #to_i simply to integer otherwise it is a string (starts at 0 instead of 1)

time_start = Time.now

while count < qty
    number += 1
    check   = 1
    factor  = 2
    sqr = Math.sqrt(number).floor # finds square root and then floors it to the lower whole integer
    while factor <= sqr             # use the sqrt to find prime potential
        if number % factor == 0
            check = 0
            break 
        end
    factor += 1
    end
    if check == 1
        $stderr.printf("%d ", number)
        count += 1
    end
end
puts "\n"
printf("%8.4f\n", Time.now - time_start)
